src/groupWith.js

```diff
@@ -0,0 +1,33 @@
+var _curry2 = require('./internal/_curry2');
+
+/**
+ * Takes a list and returns a list of lists where each sublist contains only equal elements.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig (a, a -> Boolean) -> [a] -> [[a]]
+ * @param {Function} fn Function :: a, a -> Boolean
+ * @param {Array} list The array to group
+ * @return {List} A list that contains sublists of equal elements,
+ *         whose concatenations is equal to the original list.
+ * @example
+ *
+ *      var groupWithLetters = R.groupWith(R.equals);
+ *      var letters = ["M", "i", "s", "s", "i", "s", "s", "i", "p", "p", "i"];
+ *      groupWithLetters(letters);
+ *      // [['M'], ['i'], ['s', 's'], ['i'], ['s', 's'], ['i'], ['p', 'p'], ['i']]
```

Comment:
I think an example that uses `R.equals` is nice since it might be a little more straightforward. I don't feel super strongly about this, though.